zeppelin/hdi_add_zeppelin.sh

```diff
@@ -3,127 +3,57 @@
 HEADNODE_HOST_FQDN=
-#----------------------------------------------------------------
-# PARSE AND PROCESS COMMAND LINE PARAMETERS
-#----------------------------------------------------------------
-process_arguments() 
-{
-    while :; do
-        case $1 in
-            -p|--password)
-                if [ -n "$2" ]; then
-                    AMBARI_PASSWORD=$2
-                    shift
-                else
-                    printf '[ERROR] -p requires non-empty ambari admin user password.' >&2
-                    print_usage
-                    exit 1
-                fi
-                ;;
-            -c|--clustername)
-                if [ -n "$2" ]; then
-                    CLUSTER=$2
-                    shift
-                else
-                    printf '[ERROR] -c requires non-empty DNS name of HBase cluster.' >&2
-                    print_usage
-                    exit 1
-                fi
-                ;;
-            -h|--hostname)
-                if [ -n "$2" ];then
-                    HEADNODE_HOST_FQDN=$2
-                    shift
-                else
-                    printf '[ERROR] -h requires non-empty head node host FQDN.' >&2
-                    print_usage
-                    exit 1
-                fi
-                ;;
-	    -u|--username)
-	        if [ -n "$2" ]; then
-		    AMBARI_USER = $2
-		    shift
-		else
-		    printf '[ERROR] -u requires non-empty ambari admin user name.' >&2
-		    print_usage
-		    exit 1
-		fi
-		;;
-            *)
-                break
-        esac
-        shift
-    done
+checkHostNameAndSetClusterName() {
+    HEADNODE_HOST_FQDN=$(hostname -f)
+    echo "HEADNODE_HOST_FQDN=$HEADNODE_HOST_FQDN. Lower case: ${HEADNODE_HOST_FQDN,,}"
+    if ! [[ $HEADNODE_HOST_FQDN == hn0* ]]; then 
```

Comment:
because of overprovisioning, headnode0 might not even exist. What version of the Hue script were you referring to?

Reply:
we do not overprovision headnode afaik

Reply:
added code to guard against possible overprovisoning